exec/java-exec/src/main/java/org/apache/drill/exec/server/Drillbit.java

```diff
@@ -207,6 +210,21 @@ private void javaPropertiesToSystemOptions() {
   }
 
   /**
+   * If cluster version is the same as default, update option to current drillbit version.
+   */
+  private void checkAndUpdateClusterVersionOption(OptionManager optionManager, DrillbitEndpoint drillbitEndpoint) {
+    OptionValue versionOption = optionManager.getOption(ExecConstants.CLUSTER_VERSION);
+    OptionValidator validator = SystemOptionManager.getValidator(ExecConstants.CLUSTER_VERSION);
+    if (versionOption.equals(validator.getDefault())) {
+      optionManager.setOption(OptionValue.createOption(
```

Comment:
I don't think using a system option is the right way to detect version conflicts; cluster wide upgrade and restart will cause all drillbits to generate a warning because system options are persisted.